haproxy/files/update_ocsp

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+#
+# (c) 2015 SitePoint Pty Ltd
+# Maintainer: Adam Bolte
+#
+# Add HAProxy OCSP Stapling support based on the Mozilla instructions:
+# https://wiki.mozilla.org/Security/Server_Side_TLS#Haproxy
+
+
+import argparse
+import os
+import re
+import sys
+import tempfile
+from pprint import pprint
+from subprocess import PIPE, Popen, check_call
+
+ocsp_file_ext = '.pem.ocsp'
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-v', '--verbose', action='store_true')
+parser.add_argument('OCSP_OUT_DIR', help="Where to write the OCSP files to.")
+parser.parse_args()
+args = parser.parse_args()
+ocsp_out_dir = args.OCSP_OUT_DIR
+verbose = args.verbose
+
+# Import the Salt client library
+try:
+    import salt.client
+except ImportError as e:
+    sys.stderr.write(
+        "Failed to load the Salt Stack client library.\n" + \
+        "Try installing the salt-common package.\n"
+    )
+    sys.exit(1)
+
+# Create a temporary directory.
+td = tempfile.mkdtemp(prefix='update-ocsp-')
+
+# Create a local client Caller object
+caller = salt.client.Caller()
+
+cert_files = {}
+
+# Make calls with the cmd method
+try:
+    results = caller.function('pillar.get', 'ssl')
+except Exception as e:
+    sys.stderr.write("Failure:\n%r\n" % e)
+    sys.exit(1)
+
+
+for cert in results:
+    cert_files.update({cert: {}})
+
+    results[cert].update({'ocsp': ''})
+    for cert_type in ('intermediate', 'ca', 'certificate', 'ocsp'):
+
+        if cert_type in results[cert]:
+            try:
+                os.mkdir(os.path.join(td, cert), 0700)
+            except OSError as e:
+                pass
+
+            # Inconsistent Pillar data structure. 'ca' should be treated
+            # in the same way 'intermediate' certificates are treated.
+            if cert_type == 'ca':
+                cert_type = 'intermediate'
+                write_flag = os.O_CREAT | os.O_APPEND
+                write_mode = 'a'
+            else:
+                write_flag = os.O_CREAT | os.O_WRONLY
+                write_mode = 'w'
+
+            if cert_type != 'ocsp':
+                file_name = os.path.join(td, cert, cert_type)
+            else:
+                file_name = os.path.join(ocsp_out_dir, cert + ocsp_file_ext)
+            cert_files[cert].update({cert_type: file_name})
+
+            if cert_type in results[cert]:
+                with open(file_name, write_mode) as fh:
+                    cert_start = False
+                    for line in results[cert][cert_type].splitlines():
+                        if line == '-----BEGIN CERTIFICATE-----':
+                            cert_start = True
+                        if cert_start:
+                            fh.write(line + '\n')
+
+first_line = True
+for cert in cert_files:
```

Comment:
I _think_ this can be written like
```python
for line_number, cert in cert_files:
  if verbose and line_number != 0
    print
```
or something like that, so the `first_line` isn't needed